### DISCUSSION TITLE
C API: Get advertisement type from peripheral scan result

### DISCUSSION THREAD

**Discussion by ecshsun** [simpleble/simpleble#287] [Q&A]
Hi, I need to get the advertisement type from any advertising peripheral. The BLE specification defines these types:

- 0x00 - ADV_IND - connectable and scannable undirected advertising
- 0x01 - ADV_DIRECT_IND - connectable directed advertising
- 0x02 - ADV_SCAN_IND - scannable undirected advertising
- 0x03 - ADV_NONCONN_IND - non-connectable undirected advertising
- 0x04 - SCAN_RSP - scan response

So far I am only able to get the 'connectable' property by calling `simpleble_peripheral_is_connectable()`.

**How do I figure out if the advertising is 'scannable' and/or 'directed'?**

**Comment by kdewald** [comment]
@ecshsun thanks for reaching out!

It might be possible on certain platforms, but I don't think all of them expose this information. If you can elaborate more on your use case, we can try to find a solution for it.

Feel free to reach out to `contact@simpleble.org` and we'll get the ball rolling.
